### DISCUSSION TITLE
how to make import work

### DISCUSSION THREAD

**Discussion by RobVanN-CW** [smbache/import#93] [Q&A]
Hi, coming from python and accustomed to their 'import' functionality I hoped to make use of this package but run into problems with loading existing files with functions using this R module.

    > import::from("extra_functions.R", .all=TRUE, .directory=wd)
    Error: Failed to import ie_functions2.R due to the following error:
    Cannot assign name to different value in the given environment. Name already in use.

I checked https://github.com/DS4PS/cpp-528-spr-2021/issues/54 but the solution advised there does not solve it for me:  After clearing the console and the environment, the first import steps with a test file (from the same place, `wd`) work, but then this error pops up when loading the 'extra_functions.R' script that is importing many libraries via `import::from(lib, .all=TRUE)` . 

Is it because some of those libraries import functions with the same name?

PS. I work in Rstudio, starting from a project-environment that is the same for collaborators.

**Comment by torfason** [comment]
I don't think this error that `import` is throwing, but an error that is being thrown by R on some assignment. Googling shows that this happens in other packages as well. It also suggests that there is some recursive importing going on, which is not well supported.

You probably can't share `extra_functions.R`?  In any case it will be very hard to debug this without a `reprex::reprex()` or at least a better indication of what is going on with the functions.

Does `source("extra_functions.R")` work? If so, is there a reason why it is important to use `import`, given that you are not selecting specific functions?

**Reply by RobVanN-CW** [reply]
Thanks for the quick respone. I agree that this is not related to 'import' but maybe the experience someone has with the module would have pointed to a solution. 

I tried `source` but ran into a problem relating to setting the the directory to get the files from. This seems nicely solved by using the `.directory` parameter of `import`. 

Basically I would like to split this functions file up  in more managable chunks; in a way copying a kind of file organization I used in python. That went in the beginning not smooth either because of circular dependencies between the separated modules I had to resolve. Maybe this error is based on something similar. 

Will have to do a bit more testing; thanks again for the feedback.

Rob
